sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -128,7 +128,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
         RemoveRedundantAliases,
         RemoveNoopOperators,
         SimplifyExtractValueOps,
-        CombineConcats) ++
+        CombineConcats,
+        RewriteArithmeticFiltersOnIntOrLongColumn) ++
```

Comment:
It would be better to put this rule just before `ConstantFolding`?

Reply:
yes, I've put it before `ConstantFolding`